packages/x-date-pickers/src/DigitalClock/DigitalClock.tsx

```diff
@@ -294,6 +296,30 @@ export const DigitalClock = React.forwardRef(function DigitalClock<TDate extends
     utils.isEqual(option, valueOrReferenceDate),
   );
 
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    switch (event.key) {
+      case 'PageUp': {
+        if (listRef.current) {
+          const newIndex = getFocusedListItemIndex(listRef.current) - 5;
+          const children = listRef.current?.children;
+          (children[Math.max(0, newIndex)] as HTMLElement).focus();
+        }
+        event.preventDefault();
+        break;
+      }
+      case 'PageDown': {
+        if (listRef.current) {
+          const newIndex = getFocusedListItemIndex(listRef.current) + 5;
```

Comment:
You should be able to make those more readable:

Not sure if the following works, but you get the idea :+1: 

```suggestion
          if (!listRef.current) {
            return;
          }

          const newIndex = getFocusedListItemIndex(listRef.current) + 5;
          const children = listRef.current.children;
          const newFocusedIndex = Math.min(children.length - 1, newIndex)
          children[newFocusedIndex].focus();
```